R/shiny-recorder.R

```diff
@@ -183,14 +183,18 @@ CLIENT_WS_STATE <- enum(UNOPENED, OPEN, CLOSED)
 
 RecordingSession <- R6::R6Class("RecordingSession",
   public = list(
-    initialize = function(targetAppUrl, host, port, outputFileName) {
+    initialize = function(targetAppUrl, host, port, outputFileName, connectApiKey) {
       private$targetURL <- URLBuilder$new(targetAppUrl)
       private$targetType <- servedBy(private$targetURL)
       if (private$targetType == SERVER_TYPE$SAI) {
         stop("Recording shinyapps.io apps is not supported")
       }
       private$localHost <- host
       private$localPort <- port
+      if (private$targetType == SERVER_TYPE$RSC) {
+        message("Authenticating using provided connect_api_key.")
+        private$connectApiKey <- connectApiKey
+      }
```

Comment:
```suggestion
      if (private$targetType == SERVER_TYPE$RSC && !is.null(connectApiKey)) {
        message("Authenticating using provided connect_api_key.")
        private$connectApiKey <- connectApiKey
      }
```

Reply:
This prevents a constant message of `"Authenticating using provided connect_api_key"` whenever user tests agains a RSC instance regardless if there is a key.

Now a key is required to produced the statement (and store it).